### PR TITLE
test(inputs.solr): Rework testing

### PR DESCRIPTION
**Pull request by srebhan**
- [ ] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR reworks the testing in the Solr input plugin in preparation of an approach to #4713  and #7264. There are no plugin code changes besides moving types to an own file and adding TOML tags.

### CODE REVIEW

File: plugins/inputs/solr/solr_test.go [hunk 1]
```diff
@@ -2,197 +2,198 @@ package solr
...
+				version := parts[len(parts)-1]
+				require.True(t, strings.HasPrefix(version, "v"))
+				image += ":" + strings.TrimPrefix(version, "v")
+			}
+
+			// Start the container
+			container := testutil.Container{
+				Image:        image,
+				ExposedPorts: []string{servicePort},
+				Cmd:          []string{"solr-precreate", "main"},
```
Comment (powersj) [inline plugins/inputs/solr/solr_test.go:141]: Is this the command you want the container to start with? Could the existing entrypoint value get used instead?

**Review by powersj — COMMENTED** [review]
Thank you! Only one question for you

File: plugins/inputs/solr/solr_test.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/solr/solr_test.go:141]: Unfortunately it cannot. You need this command to fill in some data to the container. If you use the usual entry point there is no data and thus some metrics (the important ones we want to test for) are missing...

Comment (powersj) [reply plugins/inputs/solr/solr_test.go:141]: ok thanks for confirming!

**Pull request merged**